source/frontend/src/features/properties/map/acquisition/common/GenerateForm/hooks/useGenerateH0443.ts

```diff
@@ -0,0 +1,172 @@
+  const { getPersonConcept } = useApiContacts();
+  const {
+    getAcquisitionFile: { execute: getAcquisitionFile },
+  } = useAcquisitionProvider();
+
+  const {
+    getMultiplePropertiesById: { execute: getMultipleProperties },
+  } = useProperties();
+
+  const { generateDocumentDownloadWrappedRequest: generate } = useDocumentGenerationRepository();
+
+  const generateLetter = async (acquisitionFileId: number) => {
+    const file = await getAcquisitionFile(acquisitionFileId);
+    if (file) {
+      // Retrieve Propery Coordinator
+      const propertyCoordinator = file.acquisitionTeam?.find(
+        team => team.personProfileTypeCode === 'PROPCOORD',
+      );
+      const coordinatorPerson = !!propertyCoordinator?.personId
+        ? (await getPersonConcept(propertyCoordinator?.personId))?.data
+        : null;
+
+      // Retrieve Property Agent
+      const propertyAgent = file.acquisitionTeam?.find(
+        team => team.personProfileTypeCode === 'PROPAGENT',
+      );
+
+      const agentPerson = !!propertyAgent?.personId
+        ? (await getPersonConcept(propertyAgent?.personId))?.data
+        : null;
+
+      // Retrieve Properties
+      const filePropertiesIds =
+        file.fileProperties?.map(fp => fp.propertyId).filter((p): p is number => !!p) || [];
+      const properties = await getMultipleProperties(filePropertiesIds);
+
+      const owners: Api_AcquisitionFileOwner[] =
+        file.acquisitionFileOwners?.filter((x): x is Api_AcquisitionFileOwner => !!x) || [];
+      const contactOwner = owners.find(x => x.isPrimaryContact === true);
+
+      const h0443Data: H0443Data = {
+        file_name: file.fileName || '',
+        file_number: file.fileNumber || '',
+        project_number: file.project?.code || '',
+        project_name: file.project?.description || '',
+        properties: properties !== undefined ? properties.map<Property>(x => getProperty(x)) : [],
+        owner_names: owners.map<string>(x => getOwnerName(x)) || [],
+        owner_contact: contactOwner !== undefined ? getOwnerContact(contactOwner) : null,
+        property_coordinator: {
+          first_name: coordinatorPerson?.firstName || '',
+          last_name: coordinatorPerson?.surname || '',
+        },
+        property_agent: {
+          first_name: agentPerson?.firstName || '',
+          last_name: agentPerson?.surname || '',
+          phone:
+            agentPerson?.contactMethods?.find(
+              x => x.contactMethodType?.id === ContactMethodTypes.WorkPhone,
+            )?.value || '',
+        },
+      };
+
+      const generatedFile = await generate({
+        templateType: FormDocumentType.H0443,
+        templateData: h0443Data,
+      });
+      generatedFile?.status === ExternalResultStatus.Success!! &&
+        generatedFile?.payload &&
+        showFile(generatedFile?.payload);
+    }
+  };
+  return generateLetter;
+};
+
+function getOwnerName(owner: Api_AcquisitionFileOwner): string {
```

Comment:
I would like to standardize this if we can. You and I should discuss this. I think we'll have a bunch of headaches if we have multiple slightly different implementations of this kicking around.